File: chap11/run-model11-10.R
library(dplyr)
library(tidyr)
library(cmdstanr)

d <- read.csv('input/data-eg1.csv') %>% 
  pivot_wider(id_cols=c(Group, PID), names_from=Time, values_from=Y) 
Y1 <- d %>% filter(Group==1) %>% select(-Group, -PID)
Y2 <- d %>% filter(Group==2) %>% select(-Group, -PID)
T <- ncol(Y1)
N1 <- nrow(Y1)
N2 <- nrow(Y2)
data <- list(T=T, N1=N1, N2=N2, Y1=Y1, Y2=Y2)

model <- cmdstan_model('model/model11-10.stan')
fit <- model$sample(data=data, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model11-10.RDS')

File: chap11/run-model11-10.py
import pandas
import cmdstanpy

d = pandas.read_csv('input/data-eg1.csv')
d = d.pivot(index=['Group', 'PID'], columns='Time', values='Y').reset_index()
Y1 = d[d.Group == 1].drop(['Group', 'PID'], axis='columns')
Y2 = d[d.Group == 2].drop(['Group', 'PID'], axis='columns')
T = Y1.shape[1]
N1 = Y1.shape[0]
N2 = Y2.shape[0]
data = {'T':T, 'N1':N1, 'N2':N2, 'Y1':Y1, 'Y2':Y2}

model = cmdstanpy.CmdStanModel(stan_file='model/model11-10.stan')
fit = model.sample(data=data, seed=123, parallel_chains=4)
fit.save_csvfiles('output/result-model11-10')
